### PR TITLE
Sagonzal/request context

### PR DESCRIPTION
**Pull request by sangonzal**
Refactoring before adding telemetry changes. No new business logic or features added in this PR. Telmetry PR soon to follow. 

- new MsalRequest classes for each flow - all request now go through InitializeRequest()
- add service bundle
- add request context  
- renaming all occurrences of adal to msal 
- fixing unit tests after changes
- small formatting changes in places to make code more readable

### CODE REVIEW

File: src/main/java/com/microsoft/aad/msal4j/MsalAuthorizationGrant.java [hunk 1]
```diff
@@ -1,39 +0,0 @@
-// Copyright (c) Microsoft Corporation.
```
Comment (sangonzal) [inline src/main/java/com/microsoft/aad/msal4j/MsalAuthorizationGrant.java]: This was not being used anywhere

File: src/main/java/com/microsoft/aad/msal4j/MsalCallable.java [hunk 2]
```diff
@@ -1,57 +0,0 @@
-// Copyright (c) Microsoft Corporation.
```
Comment (sangonzal) [inline src/main/java/com/microsoft/aad/msal4j/MsalCallable.java]: This was not being used anywhere either

File: src/main/java/com/microsoft/aad/msal4j/ServiceBundle.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package com.microsoft.aad.msal4j;
+
+import javax.net.ssl.SSLSocketFactory;
+import java.net.Proxy;
+import java.util.concurrent.ExecutorService;
+
+class ServiceBundle {
```
Comment (sangonzal) [inline src/main/java/com/microsoft/aad/msal4j/ServiceBundle.java]: TelemetryManager will also be included in here

File: src/test/java/com/microsoft/aad/msal4j/OAuthRequestValidationTest.java [hunk 4]
```diff
@@ -222,10 +228,10 @@ public void oAuthRequest_for_acquireTokenByAsymmetricKeyCredential() throws Exce
         Assert.assertFalse(StringUtils.isEmpty(queryParams.get("client_assertion")));
 
         // to do validate scopes
-        Assert.assertEquals(SCOPES, queryParams.get("scope"));
-
-        Assert.assertEquals(CLIENT_ASSERTION_TYPE_JWT, queryParams.get("client_assertion_type"));
-        Assert.assertEquals(ON_BEHALF_OF_USE_JWT, queryParams.get("requested_token_use"));
+        //TODO for OBO tests, should query params include default scopes?
+        Assert.assertEquals( queryParams.get("scope"), SCOPES);
```
Comment (sangonzal) [inline src/test/java/com/microsoft/aad/msal4j/OAuthRequestValidationTest.java]: This test is currently failing - should OBO requests include default scopes (openid profile offline_access)? Every other request has them.

Comment (jmprieur) [reply src/test/java/com/microsoft/aad/msal4j/OAuthRequestValidationTest.java]: yes should @sangonzal

Comment (sangonzal) [reply src/test/java/com/microsoft/aad/msal4j/OAuthRequestValidationTest.java]: @jmprieur thanks for confirming. I have updated the unit test to reflect this.

**Comment by henrik-me** [comment]
I noticed that lots of stuff is prepended, appended or in between "tagged" with Msal. I would recommend getting rid of that. The namespace already says it's an msal. Can this be removed?

File: src/main/java/com/microsoft/aad/msal4j/ClientApplicationBase.java [hunk 5]
```diff
@@ -156,17 +122,25 @@ protected static void validateNotEmpty(String name, Set<String> set) {
...
-    {
+
+    public CompletableFuture<AuthenticationResult> acquireTokenByAuthorizationCode(
+            Set<String> scopes,
+            String authorizationCode,
+            URI redirectUri) {
+
         validateNotBlank("authorizationCode", authorizationCode);
         validateNotBlank("redirectUri", authorizationCode);
+        validateNotEmpty("scopes", scopes);
```
Comment (SomkaPe) [inline src/main/java/com/microsoft/aad/msal4j/ClientApplicationBase.java]: Scopes are optional here

File: src/main/java/com/microsoft/aad/msal4j/ClientApplicationBase.java [hunk 6]
```diff
@@ -180,33 +154,91 @@ protected static void validateNotEmpty(String name, Set<String> set) {
...
+                clientAuthentication,
+                new RequestContext(clientId, correlationId));
 
-        return this.acquireToken(authGrant, clientAuthentication);
+        return this.InitializeRequest(refreshTokenRequest);
     }
 
-    AuthenticationResult acquireTokenCommon(AbstractMsalAuthorizationGrant authGrant, ClientAuthentication clientAuth,
-                                            ClientDataHttpHeaders headers) throws Exception {
+    CompletableFuture<AuthenticationResult> InitializeRequest(
```
Comment (SomkaPe) [inline src/main/java/com/microsoft/aad/msal4j/ClientApplicationBase.java]: this looks more like execute request, initialize typically means prepare

File: src/main/java/com/microsoft/aad/msal4j/ClientApplicationBase.java (hunk 5 above)
Comment (sangonzal) [reply src/main/java/com/microsoft/aad/msal4j/ClientApplicationBase.java]: Done

File: src/main/java/com/microsoft/aad/msal4j/ClientApplicationBase.java (hunk 6 above)
Comment (sangonzal) [reply src/main/java/com/microsoft/aad/msal4j/ClientApplicationBase.java]: Good point, changed it.

**Comment by sangonzal** [comment]
@henrik-me I removed MSAL from most things.

**Pull request merged**